src/client/play.js

```diff
@@ -5,15 +5,37 @@ const concat = require('../transforms/binaryStream').concat
 module.exports = function (client, options) {
   client.once('success', onLogin)
 
+  const mcData = require('minecraft-data')(client.version)
+
   function onLogin (packet) {
     client.state = states.PLAY
     client.uuid = packet.uuid
     client.username = packet.username
-    client.signMessage = (message, timestamp, salt = 0) => {
+    client.signMessage = (message, timestamp, salt = 0n, isCommand, preview) => {
       if (!client.profileKeys) throw Error("Can't sign message without profile keys, please set valid auth mode")
-      const signable = concat('i64', salt, 'UUID', client.uuid, 'i64',
-        timestamp / 1000n, 'pstring', JSON.stringify({ text: message }))
-      return crypto.sign('RSA-SHA256', signable, client.profileKeys.private)
+      if (mcData.supportFeature('chainedSignature')) { // 1.19.1/1.19.2
+        const hashable = crypto.createHash('sha256').update(concat('i64', salt, 'i64', timestamp / 1000n, 'pstring', message, 'i8', 70))
```

Comment:
What is the 70 referring to here? And is this verified by the server ?

Reply:
Client adds an unsigned byte with the value of 70 before hashing. Server does the same thing when verifying. No clue why.